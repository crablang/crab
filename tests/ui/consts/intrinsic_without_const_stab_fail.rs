#![feature(intrinsics, staged_api)]
#![stable(feature = "core", since = "1.6.0")]

extern "crablang-intrinsic" {
    fn copy<T>(src: *const T, dst: *mut T, count: usize);
}

#[stable(feature = "crablang1", since = "1.0.0")]
#[crablangc_const_stable(feature = "const_intrinsic_copy", since = "1.63.0")]
#[inline]
pub const unsafe fn stuff<T>(src: *const T, dst: *mut T, count: usize) {
    unsafe { copy(src, dst, count) } //~ ERROR cannot call non-const fn
}

fn main() {}
