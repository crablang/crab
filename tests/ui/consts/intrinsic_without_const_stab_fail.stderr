error[E0015]: cannot call non-const fn `copy::<T>` in constant functions
  --> $DIR/intrinsic_without_const_stab_fail.rs:12:14
   |
LL |     unsafe { copy(src, dst, count) }
   |              ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants

error: aborting due to previous error

For more information about this error, try `crablangc --explain E0015`.
